mkosi/__init__.py

```diff
@@ -5588,24 +5665,86 @@ def make_build_dir(args: CommandLineArguments) -> None:
     mkdir_last(args.build_dir, 0o755)
 
 
+def setup_ssh(args: CommandLineArguments, root: str, do_run_build_script: bool, for_cache: bool) -> Optional[TextIO]:
+    if do_run_build_script or for_cache or not args.ssh:
+        return None
+
+    if args.distribution in (Distribution.debian, Distribution.ubuntu):
+        unit = "ssh"
+    else:
+        unit = "sshd"
+
+    run(["systemctl", "--root", root, "enable", unit])
+
+    f: TextIO = cast(
+        TextIO,
+        tempfile.NamedTemporaryFile(
+            mode="w+", prefix=".mkosi-", encoding="utf-8", dir=os.path.dirname(args.output_sshkey)
+        ),
+    )
+
+    with complete_step("Generating SSH keypair"):
+        # Write a 'y' to confirm to overwrite the file.
+        run(["ssh-keygen", "-f", f.name, "-N", args.password or ""], input=f"y\n", text=True)
+
+    copy_file(f"{f.name}.pub", os.path.join(root, "root/.ssh/authorized_keys"))
+    os.remove(f"{f.name}.pub")
+
+    os.chmod(os.path.join(root, "root/.ssh/authorized_keys"), 0o600)
+
+    return f
+
+
+def setup_network_veth(args: CommandLineArguments, root: str, do_run_build_script: bool, for_cache: bool) -> None:
+    if do_run_build_script or for_cache or not args.network_veth:
+        return
+
+    network_file = os.path.join(root, "etc/systemd/network/80-mkosi-network-veth.network")
+    with open(network_file, "w") as f:
+        f.write(
+            dedent(
+                """\
+                [Match]
+                Virtualization=qemu
```

Comment:
This setting breaks networkd for me in the Debian build - unless I remove it, ens2 is ignored.

```
root@localhost:~# networkctl status ens2
● 2: ens2                                                           
                      
                     Link File: /usr/lib/systemd/network/99-defau
lt.link                  
                  Network File: n/a                              
                         
                          Type: ether                            
                         
                         State: off (unmanaged)                  
                         
             Alternative Names: enp0s2                           
                         
                          Path: pci-0000:00:02.0                 
                         
                        Driver: e1000                            
                         
                        Vendor: Intel Corporation                
                         
                         Model: 82540EM Gigabit Ethernet Controll
er (QEMU Virtual Machine)
                    HW Address: 52:54:00:12:34:56                
                         
                           MTU: 1500 (min: 46, max: 16110)       
                         
                         QDisc: noop                             
                         
  IPv6 Address Generation Mode: eui64                            
                         
          Queue Length (Tx/Rx): 1/1                              
                         
              Auto negotiation: yes                              
                         
                         Speed: 1Gbps                            
                         
                        Duplex: full                             
                         
                          Port: tp
```

Reply:
Ah, it needs to be Virtualization=kvm

```
root@localhost:~# systemd-detect-virt
kvm
```